pkg/resource/repository/hook.go

```diff
@@ -29,14 +33,108 @@ func (rm *resourceManager) setResourceAdditionalFields(
 	)
 	rm.metrics.RecordAPICall("GET", "GetLifecyclePolicy", err)
 	if err != nil {
-		if awsErr, ok := ackerr.AWSError(err); ok && awsErr.Code() == svcsdk.ErrCodeLifecyclePolicyNotFoundException {
-			ko.Spec.LifecyclePolicy = nil
-			return nil
+		if awsErr, ok := ackerr.AWSError(err); !ok || awsErr.Code() != svcsdk.ErrCodeLifecyclePolicyNotFoundException {
+			return err
 		}
-		return err
+		ko.Spec.LifecyclePolicy = nil
 	}
 	if getLifecyclePolicyResponse.LifecyclePolicyText != nil {
 		ko.Spec.LifecyclePolicy = getLifecyclePolicyResponse.LifecyclePolicyText
 	}
+
+	var listTagsForResourceResponse *svcsdk.ListTagsForResourceOutput
+	listTagsForResourceResponse, err = rm.sdkapi.ListTagsForResourceWithContext(
+		ctx,
+		&svcsdk.ListTagsForResourceInput{
+			ResourceArn: (*string)(ko.Status.ACKResourceMetadata.ARN),
+		},
+	)
+	rm.metrics.RecordAPICall("GET", "ListTagsForResource", err)
+	if err != nil {
+		return err
+	}
+	if listTagsForResourceResponse.Tags != nil {
+		tags := make([]*v1alpha1.Tag, 0, len(listTagsForResourceResponse.Tags))
+		for _, tag := range listTagsForResourceResponse.Tags {
+			tags = append(tags, &v1alpha1.Tag{
+				Key:   tag.Key,
+				Value: tag.Value,
+			})
+		}
+		ko.Spec.Tags = tags
+	}
```

Comment:
Two things.. this isn't handling pagination. Yes, I know the default number of returned tags is 100 and it's unlikely anyone is going to use more than 100 tags on a repository, but you never know (and that bug would be quite subtle and hard to debug if someone were to run into it).

Secondly, consider creating a separate helper function that calls `ListTagsForResource` and returns a `[]*v1alpha1.Tag` structs. Then here, you would do something like:

```go
ko.Spec.Tags = getRepositoryTags((*string)(ko.Status.ACKResourceMetadata.ARN))
```

Reply:
> This isn't handling pagination. Yes, I know the default number of returned tags is 100 and it's unlikely anyone is going to use more than 100 tags on a repository, but you never know (and that bug would be quite subtle and hard to debug if someone were to run into it).

Unfortunately `ListTagsForResource` call doesn't support pagination...

> Secondly, consider creating a separate helper function that calls ListTagsForResource and returns a []*v1alpha1.Tag structs. Then here, you would do something like:

++

Reply:
> > This isn't handling pagination. Yes, I know the default number of returned tags is 100 and it's unlikely anyone is going to use more than 100 tags on a repository, but you never know (and that bug would be quite subtle and hard to debug if someone were to run into it).
> 
> Unfortunately `ListTagsForResource` call doesn't support pagination... I believe this API returns all the tags at ones and have probably limited the number of tags you can attach to a resource

**sigh**. Yet another AWS API inconsistency :) OK, thanks!